base/strings/search.jl

```diff
@@ -215,7 +242,7 @@ function _searchindex(s::ByteArray, t::ByteArray, i::Integer)
 
             # match found
             if j == n - 1
-                return i+1
+                return i+f_s
```

Comment:
should stay as `i+1` since you are doing 1-based index calculations.

Reply:
not I think the returned index here refers to the actual index you want to use for a offset array